components/sup/src/config.rs

```diff
@@ -30,6 +32,32 @@ use topology::Topology;
 
 static LOGKEY: &'static str = "CFG";
 
+/// The Static Global Configuration.
+///
+/// This sets up a raw pointer, which we are going to transmute to a Box<Config>
+/// with the first call to gcache().
+static mut CONFIG: *const Config = 0 as *const Config;
+
+/// Store a configuration, for later use through `gconfig()`.
+///
+/// MUST BE CALLED BEFORE ANY CALLS TO `gconfig()`.
+pub fn gcache(config: Config) {
```

Comment:
I presume that if we ever had to update the `Config`, we'd call this again with a new copy (I noticed that `.gconfig()` doesn't return a `mut` borrow)?